### DISCUSSION TITLE
Use explicit button slots

### DISCUSSION THREAD

**Discussion by superlou** [zauberzeug/nicegui#1524] [Q&A]
### Description

I am trying to use the prepend slot on a Quasar Input to add a button at the left of the input. It should look like this:

![image](https://github.com/zauberzeug/nicegui/assets/709695/bda966b7-4d7d-4fbb-a4c5-4dfa9fe2cd78)

From the examples, using the with statement on the input seems to put all items into append. In the above image, I manually added the button as a template:

`self.input.add_slot("prepend", "<q-btn flat dense icon='edit_note'/>")`

However, I can't figure out how to wire `on_click` from inside the template. Is there a more niceui-esque approach that would let me specify which slots in the with statement?

**Comment by superlou** [comment]
I think I figured it out. You add the new slot, then you can use it with Python's `with`. For example:

```py
with ui.input().classes("col-2") as input_field:
    self.input = input_field
    self.input.props("outlined")
    
    prepend = input_field.add_slot('prepend')

    with prepend:
        if self.observe_fn:
            ui.button(icon="edit_note", on_click=self.observe) \
                .props("flat dense")
```

Is this common enough to warrant an example in the docs?

**Comment by rodja** [comment]
Normally we would simply use `ui.row` with a button and an input to achieve this. But using a slot is also possible. In general we are very short on documentation about slots. A good place would be below [generic events in the docs](https://nicegui.io/documentation#generic_events). Would you like to start something with this kind of example?

**Comment by superlou** [comment]
That seems like a good place. I'm pretty ignorant to the right way to do things in nice-gui, but I'll try to come up with something unless someone beats me to it.
